Cubical/Categories/LocallySmall/Displayed/Constructions/Reindex/Base.agda

```diff
@@ -0,0 +1,357 @@
+
+        reind' : {a b : Dob} {f g : D.Hom[ a , b ]} (p : f Eq.≡ g)
+            {aᴰ : Dobᴰ a} {bᴰ : Dobᴰ  b }
+          → Dᴰ.Hom[ f ][ aᴰ , bᴰ ]
+          → Dᴰ.Hom[ g ][ aᴰ , bᴰ ]
+        reind' p = Eq.transport Dᴰ.Hom[_][ _ , _ ] p
+
+        reind≡reind' : ∀ {a b : Dob} {f g : D.Hom[ a , b ]}
+          {p : f ≡ g} {e : f Eq.≡ g}
+          {aᴰ : Dobᴰ a} {bᴰ : Dobᴰ b }
+          → (fᴰ : Dᴰ.Hom[ f ][ aᴰ , bᴰ ])
+          → Dᴰ.reind p fᴰ ≡ reind' e fᴰ
+        reind≡reind' {p = p}{e} fᴰ =
+          subst {x = Eq.pathToEq p}
+            (λ e → Dᴰ.reind p fᴰ ≡ reind' e fᴰ)
+            (isPropEqHom _ _)
+            lem
+          where
+          lem : Dᴰ.reind p fᴰ ≡ reind' (Eq.pathToEq p) fᴰ
+          lem =
+            sym $ Eq.eqToPath $
+              Eq.transportPathToEq→transportPath
+                Dᴰ.Hom[_][ _ , _ ] p fᴰ
+
+      reindexEq : Categoryᴰ C (λ c → Dobᴰ (F.F-ob c)) _
+      reindexEq =
+        redefine-idᴰ-⋆ᴰ reindex
+          (λ {x}{xᴰ} →
+            reind' F-id' Dᴰ.idᴰ ,
+            (reind≡reind' Dᴰ.idᴰ))
+          (λ fᴰ gᴰ →
+            reind' (F-seq' _ _) (fᴰ Dᴰ.⋆ᴰ gᴰ) ,
+            reind≡reind' (fᴰ Dᴰ.⋆ᴰ gᴰ))
+
+  module _ {Dᴰ-ℓ Dobᴰ DHom-ℓᴰ}
+    (Dᴰ : SmallFibersCategoryᴰ D Dᴰ-ℓ Dobᴰ DHom-ℓᴰ) where
+    private
+      module Dᴰ = Categoryᴰ Dᴰ
+    -- Reindexing preserves smallfiberedness
+    reindexSF :
+      SmallFibersCategoryᴰ C _
+        (λ c → Dobᴰ (F.F-ob c))
+        _
+    reindexSF = reindex Dᴰ
+
+    open Functorᴰ
+    πSF : Functorᴰ F reindexSF Dᴰ
+    πSF = π Dᴰ
+
+    module _
+      (F-id' : {x : Cob} → D .id {x = F.F-ob x} Eq.≡ F.F-hom (C .id))
+      (F-seq' : {x y z : Cob}
+        (f : C.Hom[ x , y ]) (g : C.Hom[ y , z ]) →
+        (F.F-hom f) D.⋆ (F.F-hom g) Eq.≡ F.F-hom (f C.⋆ g))
+      where
+
+      reindexEqSF : SmallFibersCategoryᴰ C _ (λ c → Dobᴰ (F.F-ob c)) _
+      reindexEqSF = reindexEq Dᴰ F-id' F-seq'
+
+module _
+  {C : Category Cob CHom-ℓ}
+  {D : Category Dob DHom-ℓ}
+  (F : Functor C D)
+  where
+  private
+    module C = CategoryNotation C
+    module D = CategoryNotation D
+    module F = FunctorNotation F
+
+  module _ (Dᴰ : Categoryᴰ D Dobᴰ DHom-ℓᴰ) where
+    private
+      module Dᴰ = Categoryᴰ Dᴰ
+
+    -- Rather than acting with a reind on idᴰ and _⋆ᴰ_,
+    -- this version of reindexing stores both of the terms
+    -- that would have appeared as arguments to reind.
+    -- That is, the path and the original (untransported)
+    -- displayed hom are stored
+    --
+    -- Keeping both of these arguments around but not actually
+    -- performing the transport lets us think of the morphisms
+    -- in this displayed category as syntactic descriptions of
+    -- a reind, rather than a semantic reind
+    -- In this way, the morphisms here can be treated like deferred
+    -- transorts, and we can maximally postpone the katabasis
+    -- into transport hell
+    --
+    -- See the fiber examples below
```

Comment:
what example? Is this construction used anywhere? And can we come up with a better name?